### DISCUSSION TITLE
Tracking: current efforts to implement DreamBerd

### DISCUSSION THREAD

**Discussion by gabrielchl** [TodePond/GulfOfMexico#183] [General]
5 Feb 2024 update:
first (?) successful implementation has been given to us by @vivaansinghvi07, our lord and saviour
Repo / Fork: [vivaansinghvi07/dreamberd-interpreter](https://github.com/vivaansinghvi07/dreamberd-interpreter)

---

thought it'd be good to be able to keep track of all efforts to implement dreamberd in one place
especially seeing that more of them are likely to pop up in the coming days

- Python compiler to Javascript:
Repo / Fork: [Bytestorm5/DreamBerd](https://github.com/Bytestorm5/DreamBerd/tree/main)
Tracking PR: #74
- Java parser and runner:
Repo / Fork: [CorruptionHades/DreamBerdCompiler](https://github.com/CorruptionHades/DreamBerdCompiler/tree/master)
Tracking Discussion: #173
- Generator for parsers for any language written in Javascript:
Repo / Fork: [TodePond/MotherTode](https://github.com/TodePond/MotherTode)
Tracking Issue (rewrite in progress): TodePond/MotherTode#26
- Dreamberd.rs - Interpreter, written in Rust btw:
Repo / Fork: [PokeJofeJr4th/dreamberd-rs](https://github.com/PokeJofeJr4th/dreamberd-rs)

**Comment by TodePond** [comment]
ultra-long-way-off, but I'm also building a flexible parser generator that should do the trick: https://github.com/TodePond/MotherTode/issues/26

**Comment by PokeJofeJr4th** [comment]
Mine is somewhat far enough along that I can make it public
https://github.com/PokeJofeJr4th/dreamberd-rs

**Comment by gabrielchl** [comment]
udpate:
- added [PokeJofeJr4th/dreamberd-rs](https://github.com/PokeJofeJr4th/dreamberd-rs)
- also watching [itsamedood/DreamBerd-Interpreter](https://github.com/itsamedood/DreamBerd-Interpreter), will add to the list when there's activity

**Comment by vivaansinghvi07** [comment]
I'm currently working on an interpreter for this language which aims to get all features noted in the README, with the exception of a select few like DB3X, ABI (and etc.), and some other small things

**Comment by vivaansinghvi07** [comment]
Update: [here it is](https://github.com/vivaansinghvi07/dreamberd-interpreter)

**Reply by TodePond** [reply]
DOES IT WORK? :O

**Reply by vivaansinghvi07** [reply]
For the most part! It's a little buggy, and could use better error handling and some fixes, but I tried to stay true to the features defined in the README (to some extent)

**Reply by vivaansinghvi07** [reply]
I also am yet to write tests and publish it to PyPi for easier installation

**Reply by gabrielchl** [reply]
wot, i thought i'd never see this day lol (added to list xd)

**Reply by TodePond** [reply]
thats amazing! cant wait to try it

**Reply by vivaansinghvi07** [reply]
Just uploaded a beta version to PyPi, check out the README for installation instructions

**Reply by TodePond** [reply]
oh my god it works

![image](https://github.com/TodePond/DreamBerd/assets/15892272/f91ed64d-9f97-4532-ad9b-8d8ba1264690)

@vivaansinghvi07 I'll send you an email!

**Comment by TodePond** [comment]
Update for everyone: Yes! @vivaansinghvi07 has built enough to secure the first bounty. I've agreed to pay £99 to their chosen charity, and I'll make a full announcement around the 16th February.

![Screenshot_20240206-081354](https://github.com/TodePond/DreamBerd/assets/15892272/a0bf1fd2-638c-42bc-80bb-330c709bce82)

**Reply by vivaansinghvi07** [reply]
Thank you! I’m going to see if I can implement the other features in the ((parentheses())) discussion (I already have public `const const const`, implicit strings, and white spaced parentheses), as well as DB3X (although that I imagine would be pretty hard). Could you elaborate more on what the ECMAScript requirement is?

**Comment by James-HoneyBadger** [comment]
Here's my implementation of "Gulf of Mexico - The Perfect Programming Language"

https://github.com/James-HoneyBadger/GulfOfMexico
